feat: add more documentation

albacrespi opened the pull request:
Jessica needs this branch in main to develop further all the documentation.

albacrespi committed:
Added image

Added image to go with documentation

Signed-off-by: albacrespi <acrespi001@dundee.ac.uk>

albacrespi committed:
Update introduction.rst

Added a general description of SACRO-ML.

Signed-off-by: albacrespi <acrespi001@dundee.ac.uk>

albacrespi committed:
Update introduction.rst

Added extra line on when sacroml can be used

Signed-off-by: albacrespi <acrespi001@dundee.ac.uk>

albacrespi committed:
Fix introduction.rst

Fix irrelevant sentence.

Signed-off-by: albacrespi <acrespi001@dundee.ac.uk>

albacrespi committed:
Create Glossary.rst

Added vocabulary and definitions mainly from GRAIMatter.

Signed-off-by: albacrespi <acrespi001@dundee.ac.uk>

albacrespi committed:
Added Worst Case diagram

Signed-off-by: albacrespi <acrespi001@dundee.ac.uk>

albacrespi committed:
Update worst_case.rst

Added explanation and diagram to Worst Case attack

Signed-off-by: albacrespi <acrespi001@dundee.ac.uk>

albacrespi committed:
Added more text to intro

albacrespi committed:
Added extra explanation introduction. Fix typos. Added image.
Added autolabel functionality.

albacrespi committed:
Fix style, added and modified some definitions.

albacrespi committed:
Merge branch 'main' into 329-add-more-documentation

Signed-off-by: albacrespi <acrespi001@dundee.ac.uk>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rpreen reviewed (changes requested):
Please add "SOM" to "ignore-words-list" in `pyproject.toml` because it's getting picked up as a spelling error when it's used here as an abbreviation.

albacrespi committed:
Bug fix

Added "SOM" in "ignore-words-list" in pyproject.toml because it was getting picked up as spelling error when it's an abbreviation.

albacrespi committed:
Merge branch '329-add-more-documentation' of https://github.com/AI-SDC/SACRO-ML into 329-add-more-documentation

jim-smith commented:
This is failing tests with an 'unclosed array' error during setup

jim-smith commented on pyproject.toml:
@albacrespi needs  comma after "SOM" I think?

albacrespi committed:
bug fix

added comma
